src/tasks/installers/common-tasks.ts

```diff
@@ -94,6 +96,42 @@ export function createEclipseCheCluster(flags: any, kube: KubeHelper): Listr.Lis
   }
 }
 
+export function updateEclipseCheCluster(flags: any, kube: KubeHelper, command: Command): Listr.ListrTask {
+  return {
+    title: `Update the Custom Resource of type ${CHE_CLUSTER_CRD} in the namespace ${flags.chenamespace}`,
+    enabled: () => !!flags[CHE_OPERATOR_CR_PATCH_YAML_KEY],
+    task: async (ctx: any, task: any) => {
+      const cheOperatorCrPatchYamlPath = flags[CHE_OPERATOR_CR_PATCH_YAML_KEY]
+      if (existsSync(cheOperatorCrPatchYamlPath)) {
+        const crPatch: any = yaml.safeLoad(readFileSync(cheOperatorCrPatchYamlPath).toString())
+        const cheCluster = await kube.getCheCluster(flags.chenamespace)
+        if (!cheCluster) {
+          command.error(`Eclipse Che cluster CR was not found in the namespace ${flags.chenamespace}`)
+        }
+
+        ctx.CROverrides = {
+          spec: {
+            server: {},
+            auth: {}
+          }
+        }
+        ctx.CROverrides.spec.server.cheImage = ''
+        ctx.CROverrides.spec.server.cheImageTag = ''
+        ctx.CROverrides.spec.server.pluginRegistryImage = ''
+        ctx.CROverrides.spec.server.devfileRegistryImage = ''
+        ctx.CROverrides.spec.auth.identityProviderImage = ''
+
+        if (cheCluster && cheCluster.metadata.name) {
+          await kube.patchCheCluster(cheCluster.metadata.name, flags.chenamespace, crPatch, ctx)
```

Comment:
I think we don't have to introduce `ctx.CROverrides`
What about:
```
if (!crPatch.spec || !crPatch.spec.server || !crPatch.spec.server.pluginRegistryImage) {
 merge(crPatch, {spec: {server: { pluginRegistryImage: '' } } })
}
So, basically we remove images from the CR unless they are overridden in the patch file

```
and then 
` await kube.patchCheCluster(cheCluster.metadata.name, flags.chenamespace, crPatch)`

Reply:
Done.